### PR TITLE
add visibility to pending rule changes

### PR DESCRIPTION
**Pull request by hngerebara**
Fix to this [issue](https://github.com/mozilla-releng/balrog/issues/1224)

Previously when a change is made from release to release* it doesnt show up in the **localtest and cdntest** channels as shown below.
Screenshot 2020-09-12 at 2 17 43 PM

Now fixed as show below

Screenshot 2020-09-12 at 2 13 14 PM

Screenshot 2020-09-12 at 2 11 56 PM

### CODE REVIEW

**Comment by mozbhearsum** [comment]
Hi @hngerebara - thank you for this! I will look at it as soon as I can. It may not be until next week though.

**Review by mozbhearsum — CHANGES_REQUESTED** [review]
While this works for the very specific bug, I don't think it is the right fix. The root issue here is that the comparisons on line 464 and 467 only look at both `rule.scheduledChange` and `rule.scheduledChange.channel` (if scheduledChange is not null) instead of just one of them (ruleChannel is defined on line 456 as one or the other).

**Comment by hngerebara** [comment]
Ooh I see. I have made some modifications to that logic which fixes the issue

File: ui/src/views/Rules/ListRules/index.jsx [hunk 1]
```diff
@@ -452,13 +452,24 @@ function ListRules(props) {
...
+      const ruleScheduledChange =
+        rule.scheduledChange && rule.scheduledChange.channel;
+      let ruleChannel;
 
       if (ruleProduct !== productFilter) {
         return false;
       }
 
+      if (ruleScheduledChange) {
+        if (ruleScheduledChange.includes('*')) {
```
Comment (mozbhearsum) [inline ui/src/views/Rules/ListRules/index.jsx:464]: I don't think is right -- you want to check `ruleScheduledChange.channel` for this, not the entire rule.

**Review by mozbhearsum — CHANGES_REQUESTED** [review]
This still needs a bit of work. Here's how to make sure it's working correctly:
* Modify https://localhost:9000/rules/628 to change `release-localtest` to `release*`, and make the scheduled time at least a day in the future.
* Make sure the scheduled change is visibile on https://localhost:9000/rules?product=Firefox&channel=release-localtest, https://localhost:9000/rules?product=Firefox&channel=release-cdntest, and https://localhost:9000/rules?product=Firefox&channel=release, and not visible on https://localhost:9000/rules?product=Firefox&channel=beta

File: ui/src/views/Rules/ListRules/index.jsx (hunk 1 above)
Comment (hngerebara) [reply ui/src/views/Rules/ListRules/index.jsx:464]: @bhearsum `ruleScheduledChange` returns just the channel not the entire rule. I have also tested it out as you indicated and it works as it should

**Review by mozbhearsum — APPROVED** [review]

**Pull request merged**